File: solutions/dart/queen-attack/1/lib/queen_attack.dart
final class Queen {
  int row;
  int col;

  Queen(int this.row, int this.col) {
    assert(row >= 0, 'row not positive');
    assert(row <= 7, 'row not on board');
    assert(col >= 0, 'column not positive');
    assert(col <= 7, 'column not on board');
  }

  bool canAttack(Queen other) {
    return row == other.row ||
        col == other.col ||
        (row - other.row).abs() == (col - other.col).abs();
  }
}

File: solutions/erlang/queen-attack/1/src/queen_attack.erl
-module(queen_attack).

-export([can_attack/2]).

-spec can_attack({number(), number()}, {number(), number()}) -> boolean().
can_attack({X1, Y1}, {X2, Y2}) ->
    X1 =:= X2 orelse Y1 =:= Y2 orelse abs(X1 - X2) =:= abs(Y1 - Y2).
